File: dotNET/PFProSDK/src/PayPal/Payments/DataObjects/ECSetRequest.cs
#region "Imports"

using System;
using PayPal.Payments.Common ;
using PayPal.Payments.Common.Utility ;
#endregion

namespace PayPal.Payments.DataObjects
{
	/// <summary>
	/// Used for ExpressCheckout SET operation.
	/// </summary>
	/// <remarks>
	/// <seealso cref="ECGetRequest"/>
	/// <seealso cref="ECDoRequest"/>
	/// </remarks>
	public class ECSetRequest : ExpressCheckoutRequest
	{
		#region "Member variables"
		private String mReturnURL;
		private String mCancelURL;
		private String mReqConfirmShipping;
        private String mReqBillingAddress;
		private String mNoShipping;
		private String mAddrOverride;
		private String mLocalecode;
		private Currency mMaxAmt;
		private String mPageStyle;
		private String mHdrImg ;		//"cpp-headerimage"
		private String mHdrBorderColor;	//"cpp-header-border-color";
		private String mHdrBackColor;	//"cpp-header-back-color";
		private String mPayFlowColor;	//"cpp-payflow-color";
		private String mBillingType;
		private String mBA_Desc;
		private String mPaymentType;
		private String mBA_Custom;
        private String mShipToName;
        private String mAllowNote;

		// Transaction PayLater object. Has parameters like PromoCode, ProductCategory, etc.
		private PayLater mPayLater;

		#endregion

		#region "Constructor"
		/// <summary>
		/// Constructor for ECSetRequest
		/// </summary>
		/// <param name="ReturnUrl">String</param>
		/// <param name="CancelUrl">String</param>
		/// <remarks>
		/// ECSetRequest is used to set the data required for a Express Checkout SET operation.
		/// </remarks>
		/// <example>
		/// <code lang="C#" escaped="false">
		///		.............
		///		
		///		//Create the ECSetrequest object
		///		ECSetRequest SetEC = new ECSetRequest("http://www.yourwebsitereturnurl.com","http://www.yourwebsitecancelurl.com");
		///		
		///		.............
		/// </code>
		/// <code lang="Visual Basic" escaped="false">
		///		.............
		///		
		///		//Create the ECSetrequest object
		///		Dim SetEC As ECSetRequest = new ECSetRequest("http://www.yourwebsitereturnurl.com","http://www.yourwebsitecancelurl.com")
		///		
		///		.............
		/// </code>
		/// </example>
        public ECSetRequest(String ReturnUrl ,String CancelUrl) 
			: base(PayflowConstants.ACTIONTYPE_SET)
		{
			mReturnURL = ReturnUrl;
			mCancelURL = CancelUrl;
		}
		/// <summary>
		/// Constructor for ECSetRequest
		/// </summary>
		/// <param name="ReturnUrl">String</param>
		/// <param name="CancelUrl">String</param>
		/// <param name="PayLater">String</param>
		/// <remarks>
		/// ECSetRequest is used to set the data required for a Express Checkout SET operation.
		/// </remarks>
		/// <example>
		/// <code lang="C#" escaped="false">
		///		.............
		///		
		///		//Create the ECSetrequest object
		///		ECSetRequest SetEC = new ECSetRequest("http://www.yourwebsitereturnurl.com", "http://www.yourwebsitecancelurl.com", PayLater);
		///		
		///		.............
		/// </code>
		/// <code lang="Visual Basic" escaped="false">
		///		.............
		///		
		///		//Create the ECSetrequest object
		///		Dim SetEC As ECSetRequest = new ECSetRequest("http://www.yourwebsitereturnurl.com", "http://www.yourwebsitecancelurl.com", PayLater)
		///		
		///		.............
		/// </code>
		/// </example>
		public ECSetRequest(String ReturnUrl ,String CancelUrl, PayLater PayLater) 
			: base(PayflowConstants.ACTIONTYPE_SET)
		{
			mReturnURL = ReturnUrl;
			mCancelURL = CancelUrl;
			mPayLater = PayLater;
		}
		/// <summary>
		/// Constructor for ECSetRequest
		/// </summary>
		/// <param name="ReturnUrl">String</param>
		/// <param name="CancelUrl">String</param>
		/// <param name="BillingType">String</param>
		/// <param name="BA_Desc">String</param>
		/// <param name="PaymentType">String</param>
		/// <param name="BA_Custom">String</param>
		/// <remarks>
		/// ECSetRequest is used to set the data required for a Express Checkout SET operation.
		/// </remarks>
		/// <example>
		/// <code lang="C#" escaped="false">
		///		.............
		///		
		///		//Create the ECSetRequest object.
		///		ECSetRequest SetEC = new ECSetRequest("http://www.yourwebsitereturnurl.com","http://www.yourwebsitecancelurl.com",
		///		"MerchantInitiatedBilling", "Test Transaction", "any", "Something");
		///		
		///		.............
		/// </code>
		/// <code lang="Visual Basic" escaped="false">
		///		.............
		///		
		///		//Create the ECSetRequest object for Reference Transaction with Purchase.
		///		Dim SetEC As ECSetRequest = new ECSetRequest("http://www.yourwebsitereturnurl.com","http://www.yourwebsitecancelurl.com"
		///		"MerchantInitiatedBilling", "Test Transaction", "any", "Something")
		///		
		///		.............
		/// </code>
		/// </example>
		public ECSetRequest(String ReturnUrl, String CancelUrl, String BillingType, String BA_Desc, String PaymentType, String BA_Custom) 
			: base(PayflowConstants.ACTIONTYPE_SET)
		{
			mReturnURL = ReturnUrl;
			mCancelURL = CancelUrl;
			mBillingType = BillingType;
			mBA_Desc = BA_Desc;
			mPaymentType = PaymentType;
			mBA_Custom = BA_Custom;
		}
		protected ECSetRequest(String ReturnUrl, String CancelUrl, String BillingType, String BA_Desc, String PaymentType, String BA_Custom, String Action) 
			: base(PayflowConstants.ACTIONTYPE_SETBA)
		{
			mReturnURL = ReturnUrl;
			mCancelURL = CancelUrl;
			mBillingType = BillingType;
			mBA_Desc = BA_Desc;
			mPaymentType = PaymentType;
			mBA_Custom = BA_Custom;
		}
		#endregion

		#region "Properties"
		
		/// <summary>
		/// Gets or Sets the returnurl.
		/// </summary>
		/// <remarks>
		/// <para>Maps to Payflow Parameter:</para>
		/// <code>RETURNURL</code>
		/// </remarks>
		public String ReturnURL
		{
			get{return mReturnURL; }
			set{mReturnURL = value;}
		}

		/// <summary>
		/// Gets or Sets the cancelurl.
		/// </summary>
		/// <remarks>
		/// <para>Maps to Payflow Parameter:</para>
		/// <code>CANCELURL</code>
		/// </remarks>
		public String CancelURL
		{
			get{return mCancelURL; }
			set{mCancelURL = value;}
		}

		/// <summary>
		/// Gets or Sets the ReqConfirmShipping parameter.
		/// </summary>
		/// <remarks>
		/// <para>Maps to Payflow Parameter:</para>
		/// <code>REQCONFIRMSHIPPING</code>
		/// </remarks>
		public String ReqConfirmShipping
		{
			get{return mReqConfirmShipping; }
			set{mReqConfirmShipping = value;}
		}

        /// <summary>
        /// Gets or Sets the ReqBillingAddress parameter.
        /// </summary>
        /// <remarks>
        /// <para>Maps to Payflow Parameter:</para>
        /// <code>REQBILLINGADDRESS</code>
        /// </remarks>
        public String ReqBillingAddress
        {
            get { return mReqBillingAddress; }
            set { mReqBillingAddress = value; }
        }

		/// <summary>
		/// Gets or Sets the NoShipping parameter.
		/// </summary>
		/// <remarks>
		/// <para>Maps to Payflow Parameter:</para>
		/// <code>NOSHIPPING</code>
		/// </remarks>
		public String NoShipping
		{
			get{return mNoShipping; }
			set{mNoShipping = value;}
		}

		/// <summary>
		/// Gets or Sets the AddrOveride Parameter.
		/// </summary>
		/// <remarks>
		/// <para>Maps to Payflow Parameter:</para>
		/// <code>ADDROVERRIDE</code>
		/// </remarks>
		public String AddrOverride
		{
			get{return mAddrOverride; }
			set{mAddrOverride = value;}
		}

		/// <summary>
		/// Gets or Sets the LocaleCode Parameter.
		/// </summary>
		/// <remarks>
		/// <para>Maps to Payflow Parameter:</para>
		/// <code>LOCALECODE</code>
		/// </remarks>
		public String LocaleCode
		{
			get{return mLocalecode; }
			set{mLocalecode = value;}
		}

		/// <summary>
		/// Gets or Sets the MaxAmt Parameter.
		/// </summary>
		/// <remarks>
		/// <para>Maps to Payflow Parameter:</para>
		/// <code>MAXAMT</code>
		/// </remarks>
		public Currency MaxAmt
		{
			get{return mMaxAmt; }
			set{mMaxAmt = value;}
		}

		/// <summary>
		/// Gets or Sets the PageStyle Parameter.
		/// </summary>
		/// <remarks>
		/// <para>Maps to Payflow Parameter:</para>
		/// <code>PAGESTYLE</code>
		/// </remarks>
		public String PageStyle
		{
			get{return mPageStyle; }
			set{mPageStyle = value;}
		}

		/// <summary>
		/// Gets or Sets the HdrImg Parameter.
		/// </summary>
		/// <remarks>	
		/// <para>Maps to Payflow Parameter:</para>
		/// <code>cpp-header-image</code>
		/// </remarks>
		public String HeaderImage
		{
			get{return mHdrImg; }
			set{mHdrImg = value;}
		}

		/// <summary>
		/// Gets or Sets the HdrBorderColor Parameter.
		/// </summary>
		/// <remarks>
		/// <para>Maps to Payflow Parameter:</para>
		/// <code>cpp-header-border-color</code>
		/// </remarks>
		public String HeaderBorderColor
		{
			get{return mHdrBorderColor; }
			set{mHdrBorderColor = value;}
		}

		/// <summary>
		/// Gets or Sets the HdrBackColor Parameter.
		/// </summary>
		/// <remarks>
		/// <para>Maps to Payflow Parameter:</para>
		/// <code>cpp-header-back-color</code>
		/// </remarks>
		public String HeaderBackColor
		{
			get{return mHdrBackColor; }
			set{mHdrBackColor = value;}
		}

		/// <summary>
		/// Gets or Sets the PayFlowColor Parameter.
		/// </summary>
		/// <remarks>
		/// <para>Maps to Payflow Parameter:</para>
		/// <code>cpp-paflow-color</code>
		/// </remarks>
		public String PayFlowColor
		{
			get{return mPayFlowColor; }
			set{mPayFlowColor = value;}
		}
		/// <summary>
		/// Gets or Sets the Billing Type Parameter.
		/// </summary>
		/// <remarks>
		/// Sets up automated recurring billing for the customer.  The
		/// value is MerchantInitiatedBilling.
		/// <para>Maps to Payflow Parameter:</para>
		/// <code>BILLINGTYPE</code>
		/// </remarks>
		public String BillingType
		{
			get{return mBillingType; }
			set{mBillingType = value;}
		}
		/// <summary>
		/// Gets or Sets the Description Parameter.
		/// </summary>
		/// <remarks>
		/// Description of goods or services associated with the
		/// billing agreement.
		/// <para>Maps to Payflow Parameter:</para>
		/// <code>BA_DESC</code>
		/// </remarks>
		public String BA_Desc
		{
			get{return mBA_Desc; }
			set{mBA_Desc = value;}
		}
		/// <summary>
		/// Gets or Sets the Payment Type Parameter.
		/// </summary>
		/// <remarks>
		/// Type of payment you require. 
		/// <para>Maps to Payflow Parameter:</para>
		/// <code>PAYMENTTYPE</code>
		/// </remarks>
		public String PaymentType
		{
			get{return mPaymentType; }
			set{mPaymentType = value;}
		}
		/// <summary>
		/// Gets or Sets the Custom field Parameter.
		/// </summary>
		/// <remarks>
		/// Custom annotation field for your exclusive use.
		/// <para>Maps to Payflow Parameter:</para>
		/// <code>BA_CUSTOM</code>
		/// </remarks>
		public String BA_Custom
		{
			get{return mBA_Custom; }
			set{mBA_Custom = value;}
		}
        /// <summary>
        /// Gets or Sets the Ship to Name Parameter.
        /// </summary>
        /// <remarks>
        /// Custom annotation field for your exclusive use.
        /// <para>Maps to Payflow Parameter:</para>
        /// <code>SHIPTONAME</code>
        /// </remarks>
        public String ShipToName
        {
            get { return mShipToName; }
            set { mShipToName = value; }
        }
        /// <summary>
        /// Gets or Sets the Allow Note Parameter.
        /// </summary>
        /// <remarks>
        /// Custom annotation field for your exclusive use.
        /// <para>Maps to Payflow Parameter:</para>
        /// <code>ALLOWNOTE</code>
        /// </remarks>
        public String AllowNote
        {
            get { return mAllowNote; }
            set { mAllowNote = value; }
        }

		#endregion

		#region "Core functions"

		/// <summary>
		/// Generates the transaction request.
		/// </summary>
		internal override void GenerateRequest()
		{
			// This function is not called. All the
			// address information is validated and generated
			// in its respective derived classes.
			base.GenerateRequest ();

			RequestBuffer.Append(PayflowUtility.AppendToRequest(PayflowConstants.PARAM_RETURNURL, mReturnURL));
			RequestBuffer.Append(PayflowUtility.AppendToRequest(PayflowConstants.PARAM_CANCELURL, mCancelURL));
			RequestBuffer.Append(PayflowUtility.AppendToRequest(PayflowConstants.PARAM_REQCONFIRMSHIPPING, mReqConfirmShipping));
            RequestBuffer.Append(PayflowUtility.AppendToRequest(PayflowConstants.PARAM_REQBILLINGADDRESS, mReqBillingAddress));
			RequestBuffer.Append(PayflowUtility.AppendToRequest(PayflowConstants.PARAM_NOSHIPPING, mNoShipping));
			RequestBuffer.Append(PayflowUtility.AppendToRequest(PayflowConstants.PARAM_ADDROVERRIDE, mAddrOverride));
			RequestBuffer.Append(PayflowUtility.AppendToRequest(PayflowConstants.PARAM_LOCALECODE, mLocalecode));
			RequestBuffer.Append(PayflowUtility.AppendToRequest(PayflowConstants.PARAM_MAXAMT, mMaxAmt));
			RequestBuffer.Append(PayflowUtility.AppendToRequest(PayflowConstants.PARAM_PAGESTYLE, mPageStyle));
			RequestBuffer.Append(PayflowUtility.AppendToRequest(PayflowConstants.PARAM_HDRIMG, mHdrImg));
			RequestBuffer.Append(PayflowUtility.AppendToRequest(PayflowConstants.PARAM_HDRBORDERCOLOR, mHdrBorderColor));
			RequestBuffer.Append(PayflowUtility.AppendToRequest(PayflowConstants.PARAM_HDRBACKCOLOR, mHdrBackColor));
			RequestBuffer.Append(PayflowUtility.AppendToRequest(PayflowConstants.PARAM_PAYFLOWCOLOR, mPayFlowColor));
			RequestBuffer.Append(PayflowUtility.AppendToRequest(PayflowConstants.PARAM_BILLINGTYPE, mBillingType));
			RequestBuffer.Append(PayflowUtility.AppendToRequest(PayflowConstants.PARAM_BA_DESC, mBA_Desc));
			RequestBuffer.Append(PayflowUtility.AppendToRequest(PayflowConstants.PARAM_PAYMENTTYPE, mPaymentType));
			RequestBuffer.Append(PayflowUtility.AppendToRequest(PayflowConstants.PARAM_BA_CUSTOM, mBA_Custom));
            RequestBuffer.Append(PayflowUtility.AppendToRequest(PayflowConstants.PARAM_SHIPTONAME, mShipToName));
            RequestBuffer.Append(PayflowUtility.AppendToRequest(PayflowConstants.PARAM_ALLOWNOTE, mAllowNote));

			if (mPayLater != null)
			{
				mPayLater.RequestBuffer = RequestBuffer;
				mPayLater.GenerateRequest();;
			}
		}
		#endregion
	}
}

File: java/src/sdk/dataobjects/paypal/payflow/ECSetRequest.java
package paypal.payflow;

/**
 * Used for ExpressCheckout SET operation.
 * {@link ECGetRequest}
 * {@link ECDoRequest}
 */
public class ECSetRequest extends ExpressCheckoutRequest {

    private String returnUrl;
    private String cancelUrl;
    private String reqConfirmShipping;
    private String reqBillingAddress;
    private String noShipping;
    private String addrOverride;
    private String localecode;
    private Currency maxAmt;
    private String pageStyle;
    private String headerImg;           //"cpp-headerimage"
    private String headerBorderColor;   //"cpp-header-border-color";
    private String headerBackColor;     //"cpp-header-back-color";
    private String payFlowColor;        //"cpp-payflow-color";
    private String billingType;
    private String ba_Desc;
    private String paymentType;
    private String ba_Custom;
    private String shiptoName;
    private String allowNote;

    // Transaction PayLater object. Has parameters like PromoCode, ProductCategory, etc.
    private PayLater payLater;

    /**
     * Constructor for ECSetRequest
     *
     * @param ReturnUrl - String
     * @param CancelUrl - String
     *                  <p/>
     *                  ECSetRequest is used to set the data required for a Express Checkout SET operation.
     *                  </p>
     *  .............
     * <p/>
     * //Create the ECSetrequest object
     * ECSetRequest setEC = new ECSetRequest("http://www.yourwebsitereturnurl.com","http://www.yourwebsitecancelurl.com");
     * <p/>
     * .............
     * </code>
     */
    public ECSetRequest(String ReturnUrl, String CancelUrl) {
        super(PayflowConstants.PARAM_ACTION_SET);
        returnUrl = ReturnUrl;
        cancelUrl = CancelUrl;
    }

    /**
     * Constructor for ECSetRequest
     *
     * @param ReturnUrl - String
     * @param CancelUrl - String
     *                  <p/>
     *                  ECSetRequest is used to set the data required for a Express Checkout SET operation.
     *                  </p>
     *  .............
     * <p/>
     * //Create the ECSetrequest object
     * ECSetRequest setEC = new ECSetRequest("http://www.yourwebsitereturnurl.com","http://www.yourwebsitecancelurl.com");
     * <p/>
     * .............
     * </code>
     */
    public ECSetRequest(String ReturnUrl, String CancelUrl, PayLater PayLater) {
        super(PayflowConstants.PARAM_ACTION_SET);
        returnUrl = ReturnUrl;
        cancelUrl = CancelUrl;
        payLater = PayLater;

    }

    /**
     * Constructor for ECSetRequest
     *
     * @param ReturnUrl   - String
     * @param CancelUrl   - String
     * @param BillingType - String
     * @param BA_Desc     - String
     * @param PaymentType - String
     * @param BA_Custom   - String
     *                    <p/>
     *                    <p/>
     *                    ECSetRequest is used to set the data required for a Express Checkout SET operation for
     *                    Reference Transactions with Purchase.
     *                    </p>
     *  .............
     * <p/>
     * //Create the ECSetrequest object
     * ECSetRequest setEC = new ECSetRequest("http://www.yourwebsitereturnurl.com","http://www.yourwebsitecancelurl.com",
     * "MerchantInitiatedBilling", "Test Transaction", "any", "Something");
     * <p/>
     * .............
     * </code>
     */
    public ECSetRequest(String ReturnUrl, String CancelUrl, String BillingType, String BA_Desc, String PaymentType, String BA_Custom) {
        super(PayflowConstants.PARAM_ACTION_SET);
        returnUrl = ReturnUrl;
        cancelUrl = CancelUrl;
        billingType = BillingType;
        ba_Desc = BA_Desc;
        paymentType = PaymentType;
        ba_Custom = BA_Custom;

    }

    protected ECSetRequest(String ReturnUrl, String CancelUrl, String BillingType, String BA_Desc, String PaymentType, String BA_Custom, String Action) {
        super(PayflowConstants.PARAM_ACTION_SETBA);
        returnUrl = ReturnUrl;
        cancelUrl = CancelUrl;
        billingType = BillingType;
        ba_Desc = BA_Desc;
        paymentType = PaymentType;
        ba_Custom = BA_Custom;
    }

    /**
     * Gets the returnurl parameter.
     *
     * @return - String
     *  <p>Maps to Payflow Parameter: RETURNURL
     */
    public String getReturnUrl() {
        return returnUrl;
    }

    /**
     * Sets the Returnurl parameter.
     *
     * @param returnUrl - String
     *  <p>Maps to Payflow Parameter: RETURNURL
     */

    public void setReturnUrl(String returnUrl) {
        this.returnUrl = returnUrl;
    }

    /**
     * Gets the cancelUrl parameter.
     *
     * @return - String
     *  <p>Maps to Payflow Parameter: CANCELURL
     */
    public String getCancelUrl() {
        return cancelUrl;
    }

    /**
     * Sets the cancelUrl parameter.
     *
     * @param cancelUrl - String
     *  <p>Maps to Payflow Parameter: CANCELURL
     */

    public void setCancelUrl(String cancelUrl) {
        this.cancelUrl = cancelUrl;
    }

    /**
     * Gets the reqConfirmShipping parameter.
     *
     * @return - String
     *  <p>Maps to Payflow Parameter: REQCONFIRMSHIPPING
     */
    public String getReqConfirmShipping() {
        return reqConfirmShipping;
    }

    /**
     * Sets the reqBillingAddress parameter.
     *
     * @param reqBillingAddress - String
     *  <p>Maps to Payflow Parameter: REQBILLNGADDRESS
     */

    public void setReqBillingAddress(String reqBillingAddress) {
        this.reqBillingAddress = reqBillingAddress;
    }

    /**
     * Gets the reqBillingAddress parameter.
     *
     * @return - String
     *  <p>Maps to Payflow Parameter: REQBILLNGADDRESS
     */
    public String getReqBillingAddress() {
        return reqBillingAddress;
    }

    /**
     * Sets the reqConfirmShipping parameter.
     *
     * @param reqConfirmShipping - String
     *  <p>Maps to Payflow Parameter: REQCONFIRMSHIPPING
     */

    public void setReqConfirmShipping(String reqConfirmShipping) {
        this.reqConfirmShipping = reqConfirmShipping;
    }

    /**
     * Gets the noShipping parameter.
     *
     * @return - String
     *  <p>Maps to Payflow Parameter: NOSHIPPING
     */
    public String getNoShipping() {
        return noShipping;
    }

    /**
     * Sets the noShipping parameter.
     *
     * @param noShipping - String
     *  <p>Maps to Payflow Parameter: NOSHIPPING
     */

    public void setNoShipping(String noShipping) {
        this.noShipping = noShipping;
    }

    /**
     * Gets the addrOverride parameter.
     *
     * @return - String
     *  <p>Maps to Payflow Parameter: ADDOVERRIDE
     */
    public String getAddrOverride() {
        return addrOverride;
    }

    /**
     * Sets the addrOverride parameter.
     *
     * @param addrOverride - String
     *  <p>Maps to Payflow Parameter: ADDROVERRIDE
     */

    public void setAddrOverride(String addrOverride) {
        this.addrOverride = addrOverride;
    }

    /**
     * Gets the localecode parameter.
     *
     * @return - String
     *  <p>Maps to Payflow Parameter: LOCALECODE
     */
    public String getLocalecode() {
        return localecode;
    }

    /**
     * Sets the localecode parameter.
     *
     * @param localecode - String
     *  <p>Maps to Payflow Parameter: LOCALECODE
     */

    public void setLocalecode(String localecode) {
        this.localecode = localecode;
    }

    /**
     * Gets the maxAmt parameter.
     *
     * @return - currency
     *  <p>Maps to Payflow Parameter: MAXAMT
     */
    public Currency getMaxAmt() {
        return maxAmt;
    }

    /**
     * Sets the maxAmt parameter.
     *
     * @param maxAmt - Currency
     *  <p>Maps to Payflow Parameter: MAXAMT
     */

    public void setMaxAmt(Currency maxAmt) {
        this.maxAmt = maxAmt;
    }

    /**
     * Gets the pageStyle parameter.
     *
     * @return - String
     *  <p>Maps to Payflow Parameter: PAGESTYLE
     */
    public String getPageStyle() {
        return pageStyle;
    }

    /**
     * Sets the pageStyle parameter.
     *
     * @param pageStyle - String
     *  <p>Maps to Payflow Parameter: PAGESTYLE
     */

    public void setPageStyle(String pageStyle) {
        this.pageStyle = pageStyle;
    }

    /**
     * Gets the HeaderImg parameter.
     *
     * @return - String
     *  <p>Maps to Payflow Parameter: HDRIMG
     */
    public String getHeaderImg() {
        return headerImg;
    }

    /**
     * Sets the HeaderImg parameter.
     *
     * @param headerImg - String
     *  <p>Maps to Payflow Parameter: HDRIMG
     */

    public void setHeaderImg(String headerImg) {
        this.headerImg = headerImg;
    }

    /**
     * Gets the headerBorderColor parameter.
     *
     * @return - String
     *  <p>Maps to Payflow Parameter: HDRBORDERCOLOR
     */
    public String getHeaderBorderColor() {
        return headerBorderColor;
    }

    /**
     * Sets the headerBorderColor parameter.
     *
     * @param headerBorderColor - String
     *  <p>Maps to Payflow Parameter: HDRBORDERCOLOR
     */

    public void setHeaderBorderColor(String headerBorderColor) {
        this.headerBorderColor = headerBorderColor;
    }

    /**
     * Gets the headerBackColor parameter.
     *
     * @return - headerBackColor String
     *  <p>Maps to Payflow Parameter: HDRBACKCOLOR
     */
    public String getHeaderBackColor() {
        return headerBackColor;
    }

    /**
     * Sets the headerBackColor parameter.
     *
     * @param headerBackColor - String
     *  <p>Maps to Payflow Parameter: HDRBACKCOLOR
     */

    public void setHeaderBackColor(String headerBackColor) {
        this.headerBackColor = headerBackColor;
    }

    /**
     * Gets the payFlowColor parameter.
     *
     * @return - String
     *  <p>Maps to Payflow Parameter: PAYFLOWCOLOR
     */
    public String getPayFlowColor() {
        return payFlowColor;
    }

    /**
     * Sets the payFlowColor parameter.
     *
     * @param payFlowColor - String
     *  <p>Maps to Payflow Parameter: PAYFLOWCOLOR
     */

    public void setPayFlowColor(String payFlowColor) {
        this.payFlowColor = payFlowColor;
    }

    /**
     * Gets the paymentType parameter.
     *
     * @return - String
     *  <p>Maps to Payflow Parameter: BA_CUSTOM
     */
    public String getba_Custom() {
        return ba_Custom;
    }

    /**
     * Sets the paymentType parameter.
     *
     * @param ba_Custom - String
     *  <p>Maps to Payflow Parameter: BA_CUSTOM
     */
    public void setba_Custom(String ba_Custom) {
        this.ba_Custom = ba_Custom;
    }

    /**
     * Gets the paymentType parameter.
     *
     * @return - String
     *  <p>Maps to Payflow Parameter: PAYMENTTYPE
     */
    public String getpaymentType() {
        return paymentType;
    }

    /**
     * Sets the paymentType parameter.
     *
     * @param paymentType - String
     *  <p>Maps to Payflow Parameter: PAYMENTTYPE
     */
    public void setpaymentType(String paymentType) {
        this.paymentType = paymentType;
    }

    /**
     * Gets the ba_Desc parameter.
     *
     * @return - String
     *  <p>Maps to Payflow Parameter: BA_DESC
     */
    public String getba_Desc() {
        return ba_Desc;
    }

    /**
     * Sets the billingType parameter.
     *
     * @param ba_Desc - String
     *  <p>Maps to Payflow Parameter: BA_DESC
     */
    public void setba_Desc(String ba_Desc) {
        this.ba_Desc = ba_Desc;
    }

    /**
     * Gets the noShipping parameter.
     *
     * @return - String
     *  <p>Maps to Payflow Parameter: NOSHIPPING
     */
    public String getbillingType() {
        return billingType;
    }

    /**
     * Sets the billingType parameter.
     *
     * @param billingType - String
     *  <p>Maps to Payflow Parameter: BILLINGTYPE
     */
    public void setbillingType(String billingType) {
        this.billingType = billingType;
    }

    /**
     * Gets the shiptoName parameter.
     *
     * @return - string
     *  <p>Maps to Payflow Parameter: SHIPTONAME
     */
    public String getshiptoName() {
        return shiptoName;
    }

    /**
     * Sets the shiptoName parameter.
     *
     * @param shiptoName - String
     *  <p>Maps to Payflow Parameter: SHIPTONAME
     */

    public void setshiptoName(String shiptoName) {
        this.shiptoName = shiptoName;
    }

        /**
     * Gets the allowNote parameter.
     *
     * @return - string
     *  <p>Maps to Payflow Parameter: ALLOWNOTE
     */
    public String getallowNote() {
        return allowNote;
    }

    /**
     * Sets the allowNote parameter.
     *
     * @param allowNote - String
     *  <p>Maps to Payflow Parameter: ALLOWNOTE
     */

    public void setallowNote(String allowNote) {
        this.allowNote = allowNote;
    }

    protected void generateRequest() {
        // This function is not called. All the
        //address information is validated and generated
        //in its respective derived classes.
        super.generateRequest();
        super.getRequestBuffer().append(PayflowUtility.appendToRequest(PayflowConstants.PARAM_RETURNURL, returnUrl));
        super.getRequestBuffer().append(PayflowUtility.appendToRequest(PayflowConstants.PARAM_CANCELURL, cancelUrl));
        super.getRequestBuffer().append(PayflowUtility.appendToRequest(PayflowConstants.PARAM_REQCONFIRMSHIPPING, reqConfirmShipping));
        super.getRequestBuffer().append(PayflowUtility.appendToRequest(PayflowConstants.PARAM_REQBILLINGADDRESS, reqBillingAddress));
        super.getRequestBuffer().append(PayflowUtility.appendToRequest(PayflowConstants.PARAM_NOSHIPPING, noShipping));
        super.getRequestBuffer().append(PayflowUtility.appendToRequest(PayflowConstants.PARAM_LOCALECODE, localecode));
        super.getRequestBuffer().append(PayflowUtility.appendToRequest(PayflowConstants.PARAM_MAXAMT, maxAmt));
        super.getRequestBuffer().append(PayflowUtility.appendToRequest(PayflowConstants.PARAM_PAGESTYLE, pageStyle));
        super.getRequestBuffer().append(PayflowUtility.appendToRequest(PayflowConstants.PARAM_HDRIMG, headerImg));
        super.getRequestBuffer().append(PayflowUtility.appendToRequest(PayflowConstants.PARAM_HDRBORDERCOLOR, headerBorderColor));
        super.getRequestBuffer().append(PayflowUtility.appendToRequest(PayflowConstants.PARAM_HDRBACKCOLOR, headerBackColor));
        super.getRequestBuffer().append(PayflowUtility.appendToRequest(PayflowConstants.PARAM_PAYFLOWCOLOR, payFlowColor));
        super.getRequestBuffer().append(PayflowUtility.appendToRequest(PayflowConstants.PARAM_BILLINGTYPE, billingType));
        super.getRequestBuffer().append(PayflowUtility.appendToRequest(PayflowConstants.PARAM_BA_DESC, ba_Desc));
        super.getRequestBuffer().append(PayflowUtility.appendToRequest(PayflowConstants.PARAM_PAYMENTTYPE, paymentType));
        super.getRequestBuffer().append(PayflowUtility.appendToRequest(PayflowConstants.PARAM_BA_CUSTOM, ba_Custom));
        super.getRequestBuffer().append(PayflowUtility.appendToRequest(PayflowConstants.PARAM_ADDROVERRIDE, addrOverride));
        super.getRequestBuffer().append(PayflowUtility.appendToRequest(PayflowConstants.PARAM_SHIPTONAME, shiptoName));
        super.getRequestBuffer().append(PayflowUtility.appendToRequest(PayflowConstants.PARAM_ALLOWNOTE, allowNote));

        if (payLater != null) {
            payLater.setRequestBuffer(getRequestBuffer());
            payLater.generateRequest();
        }
    }
}